packages/suite/src/components/wallet/PrivacyAccount/SliderInput.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { forwardRef, useEffect, useImperativeHandle, useRef, useState } from 'react';
+import styled from 'styled-components';
+import { Input, InputProps, variables } from '@trezor/components';
+
+const LevelContainer = styled.div`
+    position: absolute;
+    right: 20px;
+    width: 68px;
+
+    .inputAddon {
+        font-size: ${variables.FONT_SIZE.SMALL};
+        font-weight: ${variables.FONT_WEIGHT.MEDIUM};
+        color: ${({ theme }) => theme.TYPE_LIGHT_GREY};
+        pointer-events: none;
+    }
+`;
+
+const Level = styled(Input)`
+    height: 42px;
+    padding: ${({ innerAddon }) => !innerAddon && '1px 12px 0 12px'};
+    border: 1.5px solid ${({ theme }) => theme.STROKE_GREY};
+    color: ${({ theme }) => theme.TYPE_GREEN};
+    font-size: ${variables.FONT_SIZE.H2};
+    text-align: center;
+
+    :disabled {
+        color: ${({ theme }) => theme.TYPE_LIGHT_GREY};
+        background: ${({ theme }) => theme.BG_LIGHT_RED};
+    }
+`;
+
+interface SliderInputProps extends Pick<InputProps, 'isDisabled' | 'innerAddon' | 'addonAlign'> {
+    value: number | '';
+    onChange: (number: number) => void;
+    min?: number;
+    max?: number;
+    className?: string;
+}
+
+export const SliderInput = forwardRef<
+    { setPreviousValue: (number: number) => void },
+    SliderInputProps
+>(({ value, onChange, min = 1, max = 100, className, ...props }, ref) => {
+    const [inputValue, setInputValue] = useState<number | ''>(value);
+
+    const previousValue = useRef(inputValue);
+
+    useImperativeHandle(
+        ref,
+        () => ({
+            setPreviousValue(number) {
+                previousValue.current = number;
+            },
+        }),
+        [],
+    );
+
+    useEffect(() => {
+        if (value !== inputValue) {
+            setInputValue(value);
+        }
+    }, [value]); // eslint-disable-line react-hooks/exhaustive-deps
+
+    const handleChange = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
+        if (target.value === '') {
+            setInputValue('');
+
+            return;
+        }
+
+        const number = Number(target.value);
+        if (Number.isNaN(number)) {
+            return;
+        }
+
+        previousValue.current = number;
+        setInputValue(number);
+    };
+
+    const handleFocus = () => {
+        setInputValue('');
+    };
+
+    const handleBlur = () => {
+        let formattedNumber = Number(inputValue);
+
+        if ((!formattedNumber || formattedNumber < min) && previousValue.current !== '') {
+            formattedNumber = previousValue.current;
+        }
+
+        if (formattedNumber > max) {
+            formattedNumber = max;
+        }
```

Comment:
It behaves strange if you put `0` in the input: 
<img width="1185" alt="image" src="https://user-images.githubusercontent.com/3729633/208447624-c3c736d5-1721-4ee4-bf29-7e6d0468ec45.png">
<img width="664" alt="image" src="https://user-images.githubusercontent.com/3729633/208447683-0b814f64-aa11-4112-addb-1eb50c5e2d20.png">



maybe something like this to fix it? 
```suggestion
        if (!formattedNumber && previousValue.current !== '') {
            formattedNumber = previousValue.current;
        }
        
         if (formattedNumber < min) {
            formattedNumber = min;
        }

        if (formattedNumber > max) {
            formattedNumber = max;
        }
```

Reply:
26d0ab862dd3c8c6c9bafc6f7a5ffc72365e1c85